magnolia-voting-app/src/main/java/com/example/votingapp/valueprovider/PollStatusValueProvider.java

```diff
@@ -0,0 +1,50 @@
+package com.example.votingapp.valueprovider;
+
+import java.util.*;
+
+import javax.jcr.*;
+
+import com.vaadin.data.*;
+
+import org.slf4j.*;
+
+/**
+ * PollStatusColumnDefinition
+ */
+public class PollStatusValueProvider implements ValueProvider<Node, String> {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(PollStatusValueProvider.class);
+
+    public static final String TO_DO = "To-do";
+    public static final String ON_GOING = "On-going";
+    public static final String DONE = "Done";
+
+    @Override
+    public String apply(Node source) {
+        try {
+            Calendar startDate = source.getProperty("startDate").getValue().getDate();
+            Calendar expiryDate = source.getProperty("expiryDate").getValue().getDate();
+            
+            return getPollStatus(startDate, expiryDate);
+
+        } catch (RepositoryException e) {
+            LOGGER.error("error", e);
+            return null;
+        }
+    }
+
+    public static String getPollStatus(Calendar startDate, Calendar expiryDate) {
```

Comment:
Why you need 'public static' method here, if only call in this class, should use 'private 'method enough.

Reply:
Because I think this method is used for a general purpose and should be reused in a class of the FA-13 or any classes in the future, I made it public static.